clients/egui/src/account/mod.rs

```diff
@@ -703,6 +685,45 @@ impl AccountScreen {
             update_tx.send(AccountUpdate::FileCreated(result)).unwrap();
         });
     }
+    fn create_file(&mut self) {
+        let mut focused_parent = self.tree.root.file.id;
+        for id in self.tree.state.selected.drain() {
+            focused_parent = id;
+        }
+        let core = self.core.clone();
+        let update_tx = self.update_tx.clone();
+        thread::spawn(move || {
+            let mut max = 1;
+
+            let focused_parent = core.get_file_by_id(focused_parent).unwrap();
+            let focused_parent = if focused_parent.file_type == FileType::Document {
+                focused_parent.parent
+            } else {
+                focused_parent.id
+            };
+            core.get_children(focused_parent)
+                .unwrap()
+                .iter()
+                .for_each(|f| {
+                    let re = Regex::new(r"^untitled-(\d+)\.\w*$").unwrap();
```

Comment:
I don't think this qualifies as a good usecase for regex. There's ways to do a lot of what I'm seeing here in core already: https://github.com/lockbook/lockbook/blob/master/libs/core/libs/shared/src/filename.rs#L66. Poke around for more examples.